components/dashboard/src/start/VerifyModal.tsx

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useState } from "react";
+import Alert, { AlertType } from "../components/Alert";
+import Modal from "../components/Modal";
+import { getGitpodService } from "../service/service";
+import PhoneInput from "react-intl-tel-input";
+import "react-intl-tel-input/dist/main.css";
+import "./phone-input.css";
+
+interface VerifyModalState {
+    phoneNumber?: string;
+    phoneNumberValid?: boolean;
+    sent?: Date;
+    message?: {
+        type: AlertType;
+        text: string;
+    };
+    token?: string;
+    verified?: boolean;
+}
+
+export function VerifyModal() {
+    const [state, setState] = useState<VerifyModalState>({});
+
+    if (!state.sent) {
+        const sendCode = async () => {
+            try {
+                await getGitpodService().server.sendPhoneNumberVerificationToken(state.phoneNumber || "");
+                setState({
+                    ...state,
+                    sent: new Date(),
+                });
+                return true;
+            } catch (err) {
+                setState({
+                    sent: undefined,
+                    message: {
+                        type: "error",
+                        text: err.toString(),
+                    },
+                });
+                return false;
+            }
+        };
+        return (
+            <Modal
+                onClose={() => {}}
+                closeable={false}
+                onEnter={sendCode}
+                title="User Validation Required"
+                buttons={
+                    <div>
+                        <button className="ml-2" disabled={!state.phoneNumberValid} onClick={sendCode}>
+                            Send Code via SMS
+                        </button>
+                    </div>
+                }
+                visible={true}
+            >
+                <Alert type="warning" className="mt-2">
+                    To use Gitpod you'll need to validate your account with your phone number. This is required to
+                    discourage and reduce abuse on Gitpod infrastructure.
+                </Alert>
+                <div className="mt-2 -mx-6 px-6 pt-4">
+                    Enter a mobile phone number you would like to use to verify your account.
+                </div>
+                {state.message ? (
+                    <Alert type={state.message.type} className="mt-4 py-3">
+                        {state.message.text}
+                    </Alert>
+                ) : (
+                    <></>
+                )}
+                <div className="mt-4">
+                    <h4>Mobile Phone Number</h4>
+                    {/* HACK: Below we are adding a dummy dom element that is not visible, to reference the classes so they are not removed by purgeCSS. */}
```

Comment:
**issue**: Well, that's not so invisible over here! 🙈 

<img width="497" alt="Screenshot 2022-08-23 at 7 10 16 PM" src="https://user-images.githubusercontent.com/120486/186208170-5d500bd1-c115-4b15-b4a6-565c75f1bc39.png">